frontend/src/pages/settings/Settings.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  Body,
+  Card,
+  CardBaseTemplate,
+  Column,
+  Form,
+  FormField,
+  Grid,
+  H2,
+  Modal,
+  P,
+  SerializedFormData,
+} from "@maykin-ui/admin-ui";
+import { FormEvent, useState } from "react";
+import { useLoaderData } from "react-router-dom";
+
+import { useSubmitAction } from "../../hooks";
+import "./Settings.css";
+import { UpdateSettingsAction } from "./settings.action";
+import { SettingsContext } from "./settings.loader";
+
+export type SettingsPageProps = React.ComponentProps<"main"> & {
+  // Props here.
+};
+
+/**
+ * Settings page
+ */
+export function SettingsPage({ children, ...props }: SettingsPageProps) {
+  const { zaaktypesShortProcess, zaaktypeChoices } =
+    useLoaderData() as SettingsContext;
+  const submitAction = useSubmitAction<UpdateSettingsAction>();
+
+  const [modalOpenState, setModalOpenState] = useState<boolean>(false);
+
+  const verkorteProcedureFormFields: FormField[] = [
+    {
+      autoComplete: "off",
+      label: "Verkorte zaaktypes",
+      name: "verkorte_zaaktypes",
+      type: "checkbox",
+      options: zaaktypeChoices.map((choice) => ({
+        value: choice.value,
+        label: choice.label,
+        defaultChecked: zaaktypesShortProcess.includes(String(choice.value)),
+      })),
+    },
+  ];
+
+  const handleSubmitPatchArchiveConfig = async (
+    _: FormEvent,
+    data: SerializedFormData,
+  ) => {
+    const verkorteZaaktypes = (data.verkorte_zaaktypes ?? []) as string[];
+    submitAction({
+      type: "PATCH-ARCHIVE-CONFIG",
+      payload: {
+        zaaktypesShortProcess: verkorteZaaktypes,
+      },
+    });
+    setModalOpenState(true);
+  };
+
+  return (
+    <CardBaseTemplate>
+      <Modal
```

Comment:
Can we add an additional ok (or so) button that that also closes the modal, probably clearer?